File: lib/features/tts/domain/tts_service.dart
/// Абстрактный интерфейс сервиса синтеза речи (§7.3 ТЗ).
abstract class TtsService {
  /// Инициализирует TTS-движок.
  ///
  /// [languageCode] — 'ru' или 'en'.
  Future<void> initialize(String languageCode);

  /// Озвучивает текст.
  ///
  /// Сообщения добавляются в очередь FIFO (§8.2 ТЗ).
  Future<void> speak(String text);

  /// Останавливает текущее озвучивание и очищает очередь.
  ///
  /// Вызывается по команде «Стоп» (§5.1).
  Future<void> stop();

  /// Устанавливает скорость речи.
  ///
  /// [rate] — от 0.5 до 2.0 (§4.1 ТЗ).
  Future<void> setRate(double rate);

  /// Устанавливает громкость.
  ///
  /// [volume] — от 0.0 до 1.0.
  Future<void> setVolume(double volume);

  /// Устанавливает голос из доступных в системе.
  Future<void> setVoice(String voiceName);

  /// Возвращает список доступных голосов для текущего языка.
  Future<List<String>> getAvailableVoices();

  /// Освобождает ресурсы TTS.
  Future<void> dispose();
}

File: src/features/tts/domain/ttsService.ts
/**
 * Абстрактный интерфейс TTS-сервиса.
 */

export interface TtsService {
  /** Озвучить текст. Добавляется в FIFO-очередь. */
  speak(text: string): Promise<void>;

  /** Остановить текущее озвучивание и очистить очередь */
  stop(): Promise<void>;

  /** Установить скорость речи (0.5–2.0) */
  setRate(rate: number): Promise<void>;

  /** Установить громкость (0.0–1.0) */
  setVolume(volume: number): Promise<void>;

  /** Установить язык */
  setLanguage(language: string): Promise<void>;

  /** Получить список доступных голосов */
  getAvailableVoices(): Promise<string[]>;

  /** Установить конкретный голос */
  setVoice(voiceId: string): Promise<void>;
}
